packages/fixie/src/voice.ts

```diff
@@ -110,13 +116,20 @@ export class VoiceSession {
   /** Warm up the VoiceSession by making network connections. This is called automatically when the VoiceSession object is created. */
   warmup(): void {
     console.log('[voiceSession] warming up');
+    this.changeState(VoiceSessionState.CONNECTING);
     this.audioStarted = false;
     this.started = false;
-    const url = this.params?.webrtcUrl ?? 'wss://wsapi.fixie.ai';
-    this.socket = new WebSocket(url);
-    this.socket.onopen = () => this.handleSocketOpen();
-    this.socket.onmessage = (event) => this.handleSocketMessage(event);
-    this.socket.onclose = (event) => this.handleSocketClose(event);
+    try {
+      const url = this.params?.webrtcUrl ?? 'wss://wsapi.fixie.ai';
+      this.socket = new WebSocket(url);
+      this.socket.onopen = () => this.handleSocketOpen();
+      this.socket.onmessage = (event) => this.handleSocketMessage(event);
+      this.socket.onclose = (event) => this.handleSocketClose(event);
+      this.changeState(VoiceSessionState.CONNECTED);
```

Comment:
this is the wrong time to go to CONNECTED, doesn't occur until onopen

Reply:
Good catch - thanks